core/src/main/scala/kafka/server/AbstractFetcherManager.scala

```diff
@@ -174,7 +174,8 @@ abstract class AbstractFetcherManager[T <: AbstractFetcherThread](val name: Stri
         fetcher.removePartitions(partitions)
       failedPartitions.removeAll(partitions)
     }
-    info(s"Removed fetcher for partitions $partitions")
+    if (partitions.nonEmpty)
```

Comment:
I do see many places we are doing logging based on certain conditions. Do you think it worths adding an API support in trait `Logging`?

Reply:
@abbccdda It's not a bad idea. This is something we struggle with particularly in the controller. I am a big fan of structured logging and it would be great to have standard logging APIs which enforce consistent conventions. If you're interested in following up on this, I'd be happy to help out.